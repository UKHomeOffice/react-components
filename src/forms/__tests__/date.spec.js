import React from 'react'
import { mount, render } from 'enzyme'
import Date from 'govuk-react-components/forms/date';

describe('The Date Component', () => {

  it('should render correctly when a date is passed in', () => {
    expect(
      render(<Date label="dob" name="dob" value="2018-01-10" />)
    ).toMatchSnapshot()
  });

  it('should render a hint when a hint is passed in', () => {
    expect(
      render(<Date label="dob" name="dob" value="2018-01-10" hint="Testing hint" />)
    ).toMatchSnapshot();
  });

  it('should render an error when an error is passed in', () => {
    expect(
      render(<Date label="dob" name="dob" value="2018-01-10" error="Testing error" />)
    ).toMatchSnapshot();
  });

})
