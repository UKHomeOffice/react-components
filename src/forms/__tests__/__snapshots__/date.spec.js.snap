// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Date Component should render a hint when a hint is passed in 1`] = `
<div
  class="form-group"
>
  <fieldset>
    <legend>
      <span
        class="form-label-bold"
      >
        dob
      </span>
      <span
        class="form-hint"
      >
        Testing hint
      </span>
    </legend>
    <div
      class="form-date"
    >
      <div
        class="form-group form-group-day"
      >
        <label
          class="form-label"
          for="dob-day"
        >
          Day
        </label>
        <input
          class="form-control"
          id="dob-day"
          max="31"
          min="1"
          name="dob-day"
          pattern="[0-9]*"
          type="number"
          value="10"
        />
      </div>
      <div
        class="form-group form-group-month"
      >
        <label
          class="form-label"
          for="dob-month"
        >
          Month
        </label>
        <input
          class="form-control"
          id="dob-month"
          max="12"
          min="1"
          name="dob-month"
          pattern="[0-9]*"
          type="number"
          value="01"
        />
      </div>
      <div
        class="form-group form-group-year"
      >
        <label
          class="form-label"
          for="dob-year"
        >
          Year
        </label>
        <input
          class="form-control"
          id="dob-year"
          max="2100"
          min="1900"
          name="dob-year"
          pattern="[0-9]*"
          type="number"
          value="2018"
        />
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`The Date Component should render an error when an error is passed in 1`] = `
<div
  class="form-group form-group-error"
>
  <fieldset>
    <legend>
      <span
        class="form-label-bold"
      >
        dob
      </span>
      <span
        class="error-message"
      >
        Testing error
      </span>
    </legend>
    <div
      class="form-date"
    >
      <div
        class="form-group form-group-day"
      >
        <label
          class="form-label"
          for="dob-day"
        >
          Day
        </label>
        <input
          class="form-control form-control-error"
          id="dob-day"
          max="31"
          min="1"
          name="dob-day"
          pattern="[0-9]*"
          type="number"
          value="10"
        />
      </div>
      <div
        class="form-group form-group-month"
      >
        <label
          class="form-label"
          for="dob-month"
        >
          Month
        </label>
        <input
          class="form-control form-control-error"
          id="dob-month"
          max="12"
          min="1"
          name="dob-month"
          pattern="[0-9]*"
          type="number"
          value="01"
        />
      </div>
      <div
        class="form-group form-group-year"
      >
        <label
          class="form-label"
          for="dob-year"
        >
          Year
        </label>
        <input
          class="form-control form-control-error"
          id="dob-year"
          max="2100"
          min="1900"
          name="dob-year"
          pattern="[0-9]*"
          type="number"
          value="2018"
        />
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`The Date Component should render correctly when a date is passed in 1`] = `
<div
  class="form-group"
>
  <fieldset>
    <legend>
      <span
        class="form-label-bold"
      >
        dob
      </span>
    </legend>
    <div
      class="form-date"
    >
      <div
        class="form-group form-group-day"
      >
        <label
          class="form-label"
          for="dob-day"
        >
          Day
        </label>
        <input
          class="form-control"
          id="dob-day"
          max="31"
          min="1"
          name="dob-day"
          pattern="[0-9]*"
          type="number"
          value="10"
        />
      </div>
      <div
        class="form-group form-group-month"
      >
        <label
          class="form-label"
          for="dob-month"
        >
          Month
        </label>
        <input
          class="form-control"
          id="dob-month"
          max="12"
          min="1"
          name="dob-month"
          pattern="[0-9]*"
          type="number"
          value="01"
        />
      </div>
      <div
        class="form-group form-group-year"
      >
        <label
          class="form-label"
          for="dob-year"
        >
          Year
        </label>
        <input
          class="form-control"
          id="dob-year"
          max="2100"
          min="1900"
          name="dob-year"
          pattern="[0-9]*"
          type="number"
          value="2018"
        />
      </div>
    </div>
  </fieldset>
</div>
`;
